### DISCUSSION TITLE
Drop of PHP7 support

### DISCUSSION THREAD

**Discussion by dixyes** [swow/swow#101] [General]
PHP7 still uses resource type for many 3rd-party facilities handle (like curl handle). To support hooking them, there's  a huge difficulty (thousand+ lines code for compatible and many strange bugs that can not even mitigate).
PHP7 have a less powerful type hinting, this causes some functions in PHP7 and 8 have different functions signature on definition, this also makes it's hard for Swow library enforces strong type model in same file for PHP7 and PHP8. 
Swow now is still a bleeding-edge technique, we believe in that our users enjoy newer styles and not having a burden on old system compatibility. Meanwhile, PHP7.4 have only less than one year official support now, So we determined to drop PHP7 support now.

对于一些第三方设施的勾柄（比如curl）， PHP7还在使用资源类型。对资源类型的支持带来了极大的困难（千行以上的代码用于兼容和一些离谱到甚至无法“缓解”的bug）；
PHP7类型标定的支持还不完善，这导致了PHP7和8的函数声明有些区别，进而导致了Swow的PHP库无法在同文件声明同时适用于PHP7和8的类型；
Swow 现在还是非常新的技术，我们相信我们的用户喜欢体验新技术并且几乎没有旧系统的兼容性包袱；同时PHP7的官方支持只剩不到一年了；所以我们决定现在抛弃PHP7的支持

**Comment by AuroraYolo** [comment]
PHP7的代码twose不是已经全部移除了吗？

**Reply by dixyes** [reply]
所以要发个discussion说明一下

**Reply by AuroraYolo** [reply]
好的，等着新特性哈 twose说在逼大招

**Comment by leihenshang** [comment]
这正式版啥时候能发布啊，急人

**Reply by AuroraYolo** [reply]
别急
